temporal-sdk/src/main/java/io/temporal/worker/tuning/WorkflowSlotInfo.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.worker.tuning;
+
+import io.temporal.api.enums.v1.TaskQueueKind;
+import io.temporal.api.workflowservice.v1.PollWorkflowTaskQueueRequest;
+import io.temporal.api.workflowservice.v1.PollWorkflowTaskQueueResponse;
+import java.util.Objects;
+import javax.annotation.Nonnull;
+
+/** Contains information about a slot that is being used to execute a workflow task. */
+public class WorkflowSlotInfo {
+  private final String workflowType;
+  private final String taskQueue;
+  private final String workflowId;
+  private final String runId;
+  private final String workerIdentity;
+  private final String workerBuildId;
+  private final boolean fromStickyQueue;
+
+  public WorkflowSlotInfo(
```

Comment:
Would suggest hiding this detail (can consider making slot infos abstract classes or interfaces or something and move construction to internal if wanted, or just change the constructor for every field in here and potentially add javadoc on this one and activity one saying not to rely on stability of constructor signature).

Reply:
Yeah, the lack of granular enough visibility specifiers is annoying. I'll use a doc comment for now and we can potentially change it later if/when we change more here.